File: 8. String to Integer (atoi)/8.cs
public class Solution {
    public int MyAtoi(string s) {
        string number = s.Trim(' ');
        int isPos = 1;
    
        double num = 0; 

        if (number == ""){
            return 0;
        }

        if (number[0] == '+' || number[0] == '-'){
            if (number[0] == '-'){
                isPos = -1;
            }
            number = number.Substring(1);
        }

        foreach (char c in number){
            if (!char.IsDigit(c)){
                break;
            }
            num = num * 10 + (c - '0');
        }

        num *= isPos;

        if (num < int.MinValue) return int.MinValue;
        if (num > int.MaxValue) return int.MaxValue;

        return (int)num;
    }
}

File: 8. String to Integer (atoi)/8.py
class Solution:
    def myAtoi(self, s: str) -> int:
        number = ''

        for i in s:
            if i == ' ' and number == '':
                continue
            elif i in ['+', '-'] and number == '':
                number += i
            elif i.isdigit():
                number += i
            else:
                break
        try:
            if 2**31 -1 < int(number):
                return 2**31 -1
            elif -2**31 > int(number): 
                return -2**31
            else:   
                return int(number)
        except:
            return 0

        


S = Solution()
print(S.myAtoi("2147483648"))  # Output: -42